src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,272 @@
+
+    componentDidUpdate(prevProps) {
+        const previousReportActionsCount = _.size(prevProps.reportActions);
+        const currentReportActionsCount = _.size(this.props.reportActions);
+        if (previousReportActionsCount === currentReportActionsCount) {
+            return;
+        }
+        this.makeStateWithReports();
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const source = _.get(attachmentItem, 'source', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        this.props.onNavigate({source: addEncryptedAuthTokenToURL(source), file});
+
+        return {
+            source,
+            file,
+        };
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} shouldShowArrow
+     */
+    toggleArrowsVisibility(shouldShowArrow) {
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
+        let page;
+        const actions = ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions), true);
+
+        /**
+         * Looping to filter out attachments and retrieve the src URL and name of attachments.
+         */
+        const attachments = [];
+        _.forEach(actions, ({originalMessage, message}) => {
+            // Check for attachment which hasn't been deleted
+            if (!originalMessage || !originalMessage.html || _.some(message, m => m.isEdited)) {
+                return;
+            }
+            const matches = [...originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA)];
+
+            // matchAll captured both source url and name of the attachment
+            if (matches.length === 2) {
+                const [originalSource, name] = _.map(matches, m => m[2]);
+
+                // Update the image URL so the images can be accessed depending on the config environment.
+                // Eg: while using Ngrok the image path is from an Ngrok URL and not an Expensify URL.
+                const source = tryResolveUrlFromApiRoot(originalSource);
+                if (source === this.state.source) {
+                    page = attachments.length;
+                }
+
+                attachments.push({source, file: {name}});
+            }
+        });
+
+        // If there's no more attachments added from fetching older messages then call this again
+        if (attachments.length === this.state.attachments.length) {
+            this.fetchMoreActions();
+            return;
+        }
+
+        const {file} = this.getAttachment(attachments[page]);
+        this.setState({
+            page,
+            attachments,
+            file,
+            isLoading: false,
+            isForwardDisabled: page === 0,
+            isBackDisabled: page === attachments.length - 1,
```

Comment:
This is weird, logic

consider if there is  only one attachment in loaded chats and there is more to load from history with back being disabled we can't view it.